### DISCUSSION TITLE
Glide presets

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#5937] [General]
Hi

Quick one, I've been using Glide presets (super useful). Most of the time I use presets to crop an image, this would give me a square crop

```
'w1024_h1024' => ['w' => 1024, 'h' => 1024, 'q' => 85],
```

When I want to just change the width and make sure the height follows automatically, I've often used this

```
'w1024_h9999' => ['w' => 1024, 'h' => 9999, 'q' => 85, 'fit' => 'contain'],
```

I'm sure this used to work, however I've noticed that this now generates an image with a height of 9999! 

If I remove `'h' => 9999` so it looks like this

```
'w1024_h9999' => ['w' => 1024, 'q' => 85, 'fit' => 'contain'],
```
Glide generates an image `1024` wide and the height appears to be in proportion - which is what I want

I'm just wondering, is this the right way to do it? Or is there `'h' => auto,` attribute or similar I should be using?

**Comment by edalzell** [comment]
So  you have something that works? If it works, it's the right way.

**Comment by jasonvarga** [comment]
If you do `'h' => 9999` it seems correct that the image would be the height you requested.

See https://glide.thephpleague.com/2.0/api/size/#fit-fit for details of the different options and what to expect.

**Comment by stephenmeehanuk** [comment]
Good stuff. Just wanted to make sure. It's working, as expected.
